### ISSUE TITLE
[Question]: Saving of non-critical broken documents.

### ISSUE THREAD

**Issue by pvasilek** [mozilla/pdf.js#16476]
Hello,

sometimes I have cosmetically broken documents that displayed fine in the default viewer with warnings like
Warning: XRef.parse - Invalid "Root" reference: "XRefEntryException: Bad (uncompressed) XRef entry: 1R".

But if I fill form in such document and trying to save, only original document data saves. So, changes not apply.
Would be glad for any advice or notes.

Regards.

**Comment by Snuffleupagus** [comment]
> sometimes I have cosmetically broken documents that displayed fine in the default viewer with warnings like
> Warning: XRef.parse - Invalid "Root" reference: "XRefEntryException: Bad (uncompressed) XRef entry: 1R".

This means that your PDF document is *corrupt*, and while most PDF viewer will do their best to parse/display such documents there's quite frankly no guarantee that everything works "correctly" in such cases.

---

Please *always* fill out the ISSUE_TEMPLATE with all requested details, since as-is this issue unfortunately isn't actionable.

**Comment by pvasilek** [comment]
Yes, it is clear the document is corrupted. But does it mean the document can not be updated with annotations (which makes the form data)? Could it be there a flag like document.canBeSaved?

**Comment by Snuffleupagus** [comment]
Please actually provide an example PDF document, as asked for, since it's *completely impossible* to improve or fix the situation here without a lot more details. As-is we have no choice but to close this issue as INCOMPLETE.

**Comment by pvasilek** [comment]
Ok, then lets close it. PDFs are the medical forms, so I can not share them in public. It was just a piece of hope to give you an idea of PDF.js improvement.

**Comment by marco-c** [comment]
Could you share a document filled with fake information so it is not sensitive?

**Comment by pvasilek** [comment]
Thank you for the reply.
Part of the report is pre-filled in standard static text form and only a few fields are left for extra information. If I will try to edit the static text in Adobe Acrobat it would fix the structure.
If I can share sample privately, probably it is the case. But I did not find a way to do that.
